src/car/car.service.ts

```diff
@@ -52,6 +52,53 @@ export class CarService {
         return carUsage;
     }
 
+    async getAllCarsUsage(): Promise<object> {
+        const monthName = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+
+        //! VERY IMPORTANT QUERY - fully check it out
+        const query = `SELECT cars.car_id, reservations.id reservation_id,car_name, car_license_plate, total_price, start_date, end_date FROM cars LEFT JOIN reservations ON cars.car_id = reservations.car_id`;
+
+        const carsData = await this.databaseService.executeQuery(query);
+
+        const allCarsUsage = carsData.rows.reduce((acc, car) => {
+            const startDate = new Date(car.start_date);
+            const endDate = new Date(car.end_date);
+            const days = Math.ceil((endDate.getTime() - startDate.getTime()) / (1000 * 3600 * 24));
+            const month = startDate.getMonth();
+
+            if (acc[car.car_id]) {
+                if (acc[car.car_id].record[month]) {
+                    acc[car.car_id].record[month].days += days;
+                    acc[car.car_id].record[month].count++;
+                } else {
+                    acc[car.car_id].record[month] = { days, count: 1 };
+                }
+            } else {
+                acc[car.car_id] = { record: { [month]: { days, count: 1 } } };
+            }
+            return acc;
+        }, {});
+
+        for (const car in allCarsUsage) {
+            for (const month in allCarsUsage[car].record) {
+                allCarsUsage[car].record[month].percentage = ((allCarsUsage[car].record[month].days / 30) * 100).toFixed(2);
+
+                allCarsUsage[car].license_plate = carsData.rows.find(row => row.car_id === Number(car)).car_license_plate;
+                allCarsUsage[car].name = carsData.rows.find(row => row.car_id === Number(car)).car_name;
+
+                allCarsUsage[car].record[monthName[month]] = allCarsUsage[car].record[month];
+                delete allCarsUsage[car].record[month];
```

Comment:
it's better to create new resulting object instead of mutating existed one, so you can avoid this

Reply:
I will refactor the changes you have mentioned and create a new PR.